### ISSUE TITLE
ADLSearch will only support 8 digits in min/max size

### ISSUE THREAD

**Issue by Sopor** [eiskaltdcpp/eiskaltdcpp#456]
If i import an ADLSearch.xml it seems that EiskaltDC++ can't handle more than 8 digits (99999999) but other clients can handle 19 digits (in this case it is only 10, 1000000001).

![image](https://user-images.githubusercontent.com/5789283/103231961-7d5ad380-4939-11eb-85f9-3d26bc28c1e7.png)

It seems that the ADLSearch will work until i change that entry. Then it will change to 99999999 and if i want to get 1000000001 back i need to edit ADLSearch.xml in an external editor.

Could you add support for more than 8 digits?

Windows 10
EiskalrDC++ 2.4.0

**Comment by tehnick** [comment]
This is not that simple as you see it. `QSpinBox` works with integer numbers (`int32`) so it supports numbers only up to:
```
2^31 = 2147483648
```
So "19 digits" support is impossible with it in a principle.

I'll see if we can use `QDoubleSpinBox` here...

**Comment by tehnick** [comment]
@Sopor 
Could you explain why do you need such big numbers in bytes if EiskaltDC++ allows to limit size in KiB, MiB and GiB?..

**Comment by tehnick** [comment]
> but other clients can handle 19 digits

Probably they cannot set size limitations in human readable format (KiB, MiB, GiB) unlike EiskaltDC++ and require full size in bytes...

**Comment by Sopor** [comment]
I don't think you can set the value to exact that value if you use KiB, MiB or GiB. The client have all to choose from but you can't use comma sign or dots.

So how do you set 1000000001 in KiB, MiB or GiB?

**Comment by tehnick** [comment]
> So how do you set 1000000001 in KiB, MiB or GiB? 

In no way. But this is not needed: users just use range as it designed and search exact files amongst found files.

I have tried to increase amount of digits in input fields to 10 and this looks ugly. WONTFIX. Sorry.

**Comment by Sopor** [comment]
How can it look ugly?

**Comment by andbo467** [comment]
Here you go. A custom QLongLongSpinBox derived from QAbstractSpinBox. It is needed for Clean.ADL.Anti-RAR releases.

Regarding that it "looks ugly" you don't need to change the appearance in gui, just let there be more numbers that can fit in the gui spinbox for those who needs it. As long as there is less then 8 numbers, the user wont see any difference.

**Comment by tehnick** [comment]
> A custom QLongLongSpinBox (and here) derived from QAbstractSpinBox.

`QDoubleSpinBox` without decimal part may be used as well. This is not a problem.

> Another approach would be to scrap the idea with QSpinBox and just use QLineEdit

Yep, this is possible too. Especially with QValidator.

> Regarding that it "looks ugly" you don't need to change the appearance in gui, just let there be more numbers that can fit in the gui spinbox for those who needs it. As long as there is less then 8 numbers, the user wont see any difference.

I'll think about this. Thanks!

**Comment by BRTPOB** [comment]
I would love for this to be implemented, specifically for the Clean.ADL.Anti-RAR releases support. I just found the client, and I've been previously using AirDC++ but the web client is lacking, and the native client only works on Windows...which requires a VM and a PITA.
